testing/src/test/scala/org/http4s/Http4sSpec.scala

```diff
@@ -99,18 +98,7 @@ trait Http4sSpec
   }
 
   def withResource[A](r: Resource[IO, A])(fs: A => Fragments): Fragments =
-    r match {
-      case Resource.Allocate(alloc) =>
-        alloc
-          .map { case (a, release) =>
-            fs(a).append(step(release(ExitCase.Completed).unsafeRunSync()))
-          }
-          .unsafeRunSync()
-      case Resource.Bind(r, f) =>
-        withResource(r)(a => withResource(f(a))(fs))
-      case Resource.Suspend(r) =>
-        withResource(r.unsafeRunSync() /* ouch */ )(fs)
-    }
+    r.use(a => IO(fs(a))).unsafeRunSync()
```

Comment:
I think this is going to provide a released `a` to the fragments when they run.  Note how the old implementation was appending a step to the returned fragments.

If we can't use this allocate backdoor anymore, we're going to have a struggle in a couple modules.

Reply:
is the allocated method gone in cs3 completely?

Reply:
if not, could we use that to append the step?

Reply:
`allocated` is still around. 

I'll go back to it and figure out a way 👍